report/4.Solution/LanuageOfCompiler.tex

```diff
@@ -0,0 +1,37 @@
+\section{The languages of the compiler}
+When creating a compiler there are three languages of relevance, as see in the tombstone diagram\ref{fig:TStoneExample}.
+
+\begin{figure}[H]
+	\centering
+	\includegraphics[width=\textwidth/2+\textwidth/4]{4.Solution/images/T-diagram.png}
+	\caption{
+		Example of Tombstone diagram showcasing an example of three languages for a given compiler.\cite{TStoneWiki}
+	}
+	\label{fig:TStoneExample}
+\end{figure}
+In the case of the HCL compiler the three languages are as follows:
+
+\begin{align*}
+input \to & HCL \\
+output \to & C++/C \\
+compiler \to & Kotlin \\
+\end{align*}
+
+\subsection{Language of the compiler}
+The compiler was written in Kotlin\cite{KotlinWebsite}, a Java based language.
```

Comment:
I am not a fan of "A java based language". Try something more along the lines of: "which is a relatively new language that runs on the java virtual machine.
It also uses some of Java's standard libraries and it kan operate with Java code"

Reply:
I can see I am an idiot: Kan to can